board.cpp

```diff
@@ -0,0 +1,205 @@
+}
+
+bool Mine::operator==(const Mine &b) const{
+    return ((xpos == b.xpos)
+        && (ypos == b.ypos))
+        && (owner == b.owner);
+}
+
+bool Mine::operator==(const Position &b) const{
+    return ((xpos == b.xpos)
+        && (ypos == b.ypos));
+}
+
+Board::~Board(){
+    delete(players);
+}
+
+bool isPositionValid(Board const& board, Position pos){ //check if pos if withing acceptable values 
+    //If the position is outside the board, return false;
+    if((pos.xpos < 1)
+    || (pos.ypos < 1)
+    || (pos.xpos > board.width)
+    || (pos.ypos > board.height)) return false;
+    //if the position is disabled, return false
+    for(int i = 0; i < board.disabledPositions.size(); i++){
+        if(pos == board.disabledPositions[i]) return false;
+    }
+    //else, return true
+    return true;
+}
+
+//gets all possible valid positions
+//mainly for bot logic
+std::vector<Position> getValidTiles(Board const& board){ 
+    std::vector<Position> validPositions;
+    for(int x = 1; x <= board.width; x++){
+        for(int y = 1; y <= board.height; y++){
+            Position pos;
+            pos.xpos = x;
+            pos.ypos = y;
+            if (isPositionValid(board, pos)){
+                validPositions.push_back(pos);
+            }
+        }
+    }
+    return validPositions;
+}
+
+int getValuesWithinRange(std::string prompt, int min, int max){ //function to input an int value within range, used for board initialization, probably could be reporopused for other uses
+    int ret = min-1;
+    while ((ret < min) || (ret > max)){
+        std::cout << prompt << std::endl;
+        std::cout << "value must be between " << min << " and " << max << std::endl;
+        std::cin >> ret;
+    }
+    return ret;
+    
+}
+//board construction
+Board createBoard(){
+    Board board;
+    bool created = false;
+    int width = NULL;
+    int height = NULL;
+    int mineCount = 0;
+    board.gameType = static_cast<gameType>(getValuesWithinRange("choose game mode (0=PVP, 1=PVE)", 0, 1)); //not sure if this is good practice
+    board.width = getValuesWithinRange("choose the width of the field", 5, 10);
+    board.height = getValuesWithinRange("choose the height of the field", 5, 10);
+    mineCount = getValuesWithinRange("choose the number of mines on the field", 3, 8);
+    board.playerCount = 2; //TODO: make value be chosen by the player once support for more than two players is added
+    board.players = new Player[board.playerCount]; //dynamic array since we don't know the number of players beforehand
+    for(int p = 0; p < board.playerCount; p++){
+        Player player;
+        player.mineCount = mineCount;
+        player.id = p;
+        if(p==1 && board.gameType == PVE){ //set player 2 to be controlled by AI, may need to be refactored when support for more than 2 players is implemented
+            player.isAI = true;
+        }
+        board.players[p] = player;
+        }
+    return board;
+}
+
+namespace printBoardAuxiliars{

```

Comment:
#nice use of a namespace for "private" stuff

#could inline implementations, though